pkg/controller/containersource/reconcile.go

```diff
@@ -83,6 +83,7 @@ func (r *reconciler) Reconcile(ctx context.Context, object runtime.Object) (runt
 
 	err = r.setSinkURIArg(ctx, source, args)
 	if err != nil {
+		r.recorder.Eventf(source, corev1.EventTypeWarning, "SetSinkURIFailed", "Failed to set Sink URI: %v", err)
```

Comment:
While this works, everywhere else the status and event are set/emitted one right after the other. Why is this one separated out?

Reply:
It's fine as-is. Maybe add a comment in `setSinkURIArg` saying that an event will be recorded outside if an error is returned.

Reply:
Done.